docs/docs/reference/singleton-types.md

```diff
@@ -0,0 +1,39 @@
+---
+layout: doc-page
+title: "Literal Singleton Types"
+---
+
+Singleton types is one of the most popular features in Scala. Literal Singleton types takes it a step further.
```

Comment:
I would cut the intro down to the most important bit, something like this:

Literal Singleton Types allows primitive literals to be used as types. For example:

```scala
val fourtyTwo: 42 = 42

val hero: "Jedi".type = "Jedi"

val trivial: true = true
```